modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/RuleList/RuleList.es.js

```diff
@@ -12,134 +12,126 @@
  * details.
  */
 
-import 'clay-button';
-
-import 'clay-label';
-
-import 'clay-dropdown';
-import ClayTooltip from 'clay-tooltip';
+import ClayLabel from '@clayui/label';
+import ClayIcon from '@clayui/icon';
+import ClayDropDown from '@clayui/drop-down';
 import {PagesVisitor} from 'dynamic-data-mapping-form-renderer';
-import Component from 'metal-component';
 import dom from 'metal-dom';
-import {EventHandler} from 'metal-events';
-import Soy from 'metal-soy';
-import {Config} from 'metal-state';
-
 import {maxPageIndex, pageOptions} from '../../util/pageSupport.es';
 import {getFieldProperty} from '../LayoutProvider/util/fields.es';
 import RulesSupport from '../RuleBuilder/RulesSupport.es';
-import templates from './RuleList.soy';
+import React from 'react';
 
 /**
  * RuleList.
- * @extends Component
+ * @extends React.Component
  */
 
-class RuleList extends Component {
-	attached() {
-		this._eventHandler = new EventHandler();
-
-		this._eventHandler.add(
-			dom.on(
-				document,
-				'mouseup',
-				this._handleDocumentMouseUp.bind(this),
-				true
-			)
-		);
-
-		ClayTooltip.init();
+class RuleList extends React.Component {
+	componentDidMount() {
+        document.addEventListener('mouseup', this._handleDocumentMouseUp.bind(this), true);
```

Comment:
We can test if this event handler is being removed or not adding a console.log on this function when the component was removed from React tree. Also, I can't find where `_handleDocumentMouseUp ` is defined on this code